src/foundry/foundry.js/canvas.d.ts

```diff
@@ -17,401 +20,498 @@
  * canvas.recenter(); // Re-center the canvas on the currently controlled Token.
  * ```
  */
-declare class Canvas {
-  constructor();
-
-  protected _dragDrop: DragDrop;
-
-  app: PIXI.Application;
-
-  stage: PIXI.Container;
-
-  hud: HeadsUpDisplay;
-
-  background: BackgroundLayer | undefined;
-
-  foreground: ForegroundLayer | undefined;
-
-  drawings: DrawingsLayer | undefined;
-
-  grid: GridLayer | undefined;
-
-  walls: WallsLayer | undefined;
 
-  templates: TemplateLayer | undefined;
+declare global {
+  class Canvas {
+    constructor();
+
+    /**
+     * A reference to the currently displayed Scene document, or null if the Canvas is currently blank.
+     * @defaultValue `null`
+     */
+    scene: InstanceType<ConfiguredDocumentClass<typeof Scene>> | null;
+
+    /**
+     * The current pixel dimensions of the displayed Scene, or null if the Canvas is blank.
+     * @defaultValue `null`
+     */
+    dimensions: Canvas.Dimensions | null;
+
+    /**
+     * A reference to the HeadsUpDisplay container which overlays HTML rendering on top of this Canvas.
+     * @defaultValue `null`
+     */
+    hud: HeadsUpDisplay | null;
+
+    /**
+     * An Array of pending canvas operations which should trigger on the next re-paint
+     * @defaultValue `[]`
+     */
+    pendingOperations: Array<[(args: any[]) => void, any, any[]]>;
```

Comment:
```suggestion
    pendingOperations: Array<[fn: (args: any[]) => void, scope: any, args: any[]]>;
```
(makes it a bit more understandable I think)